### ISSUE TITLE
kitsu does not deserialize nested includes

### ISSUE THREAD

**Issue by freality** [wopian/kitsu#533]
I'm finding that kitsu is not deserializing nested includes. Given the request `api.get('manga', { params: { include: 'characters.character' } })`, I'd expect the returned data to deserialize both manga --> characters & characters --> character. But that's not what is returned.

I hope this test illustrates the issue:
```
it("deserializes a collection of resources with nested relationships", () => {
      expect.assertions(1);
      const deserialized = deserialise({
        data: [
          {
            id: "9",
            type: "users",
            attributes: {
              name: "Josh",
            },
            relationships: {
              waifu: {
                data: {
                  id: "1",
                  type: "characters",
                },
                relationships: {
                  team: {
                    data: {
                      id: "1",
                      type: "team",
                    },
                  },
                },
              },
            },
          },
        ],
        included: [
          {
            id: "1",
            type: "characters",
            attributes: {
              name: "Genkai",
            },
          },
          {
            id: "3",
            type: "team",
            attributes: {
              name: "Team Urameshi",
            },
          },
        ],
      });

      expect(deserialized).toEqual({
        data: [
          {
            name: "Josh",
            waifu: {
              data: {
                name: "Genkai",
                id: "1",
                type: "characters",
                team: {
                  data: {
                    id: "3",
                    type: "team",
                    name: "Team Urameshi",
                  },
                },
              },
            },
            id: "9",
            type: "users",
          },
        ],
      });
    });
```

**Comment by wopian** [comment]
Hi @freality, thank you for writing out an example test case for the issue. From taking a look at the example, it does not appear to be valid JSON:API 1.0 syntax for relationships. https://jsonapi.org/format/#document-resource-object-relationships.

Valid JSON:API 1.0 syntax for nested relationships would be:

```js
const deserialized = deserialise({
  data: {
    id: '1',
    type: 'manga',
    relationships: {
      characters: {
        data: [{
          type: 'mediaCharacters',
          id: '125079'
        }]
      }
    }
  },
  included: [{
    id: '125079',
    type: 'mediaCharacters',
    relationships: {
      character: {
        data: {
          type: 'characters',
          id: '84363'
        }
      }
    }
  }, {
    id: '84363',
    type: 'characters',
    attributes: {
      canonicalName: 'Gengo Kurosaka',
      names: {
        en: 'Gengo Kurosaka',
        ja_jp: '黒坂 玄悟'
      }
    }
  }]
})
```

Which passes the tests and returns

```js
expect(deserialized).toEqual({
  data: [{
    id: '1',
    type: 'manga',
    characters: {
      data: [{
        id: '125079',
        type: 'mediaCharacters',
        character: {
          data: {
            id: '84363',
            type: 'characters',
            canonicalName: 'Gengo Kurosaka',
            names: {
              en: 'Gengo Kurosaka',
              ja_jp: '黒坂 玄悟'
            }
          }
        }
      }]
    }
  }]
})
```

**Comment by wopian** [comment]
Your example with the correct JSON:API syntax for nested relationships deserialises to the expected values (corrected the `team` relationship entry in `characters` pointing to `id: '1'` instead of `id: '3'`, as provided in the `included` array)

```diff
const deserialized = deserialise({
  data: [
    {
      id: '9',
      type: 'users',
      attributes: {
        name: 'Josh'
      },
      relationships: {
        waifu: {
          data: {
            id: '1',
            type: 'characters'
          },
-          relationships: {
-            team: {
-              data: {
-                id: '1',
-                type: 'team'
-              }
-            }
-          }
        }
      }
    }
  ],
  included: [
    {
      id: '1',
      type: 'characters',
      attributes: {
        name: 'Genkai'
      },
+      relationships: {
+        team: {
+          data: {
+            id: '3',
+            type: 'team'
+          }
+        }
+      }
    },
    {
      id: '3',
      type: 'team',
      attributes: {
        name: 'Team Urameshi'
      }
    }
  ]
})

expect(deserialized).toEqual({
  data: [
    {
      name: 'Josh',
      waifu: {
        data: {
          name: 'Genkai',
          id: '1',
          type: 'characters',
          team: {
            data: {
              id: '3',
              type: 'team',
              name: 'Team Urameshi'
            }
          }
        }
      },
      id: '9',
      type: 'users'
    }
  ]
})
```

**Comment by freality** [comment]
Thanks @wopian for looking into the issue. I'll try to update my test to better illustrate what I'm seeing. 

But, in short, I'm working on a project which must retrieve nested related documents (i.e. content --> relationship:category --> relationship:image); where some data nodes have the same relationship. In my deserialized return set, the first occurance of a nested relationship deserializes properly. But all subsequent occurrences does not deserialize the deeper nested relationship data.

I know this is hard to follow. So, I'll try to update my test case.

**Comment by wopian** [comment]
Should be fixed by #601
